library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -34,6 +34,7 @@
     private static final String[] PROJECT_BATCH_ID = {DownloadContract.Batches._ID};
     private static final String WHERE_DELETED_VALUE_IS = DownloadContract.Batches.COLUMN_DELETED + " = ?";
     private static final String[] MARKED_FOR_DELETION = {"1"};
+    private static final String SORT_BY_ID_DESCENDING = DownloadContract.Batches._ID + " DESC";
```

Comment:
take this scenario,

A download is added to the queue, it's denied
Another download is added to the queue, it's allowed to download.

FIFO causes the first download to show it's `download rule error` and then start downloading the second.

Perhaps the question is whether a denied download should then block the rest of the queue?